Add supply chain catalog

SantiagoTorres opened the pull request:
This is intended to be the starting point of the supply-chain security project.
For discussion about this, see #224

lumjjb commented:
I'm thinking that we could create a folder for all similar resources, so as to not end up with a ton of folders at the root level. But this can be done later as well. Not a pressing matter.

SantiagoTorres commented:
Hi @lumjjb, I reviewed the directory structure and it appears to me it's reasonable (i.e., `supply-chain-security/compromises`). Should I move this under a directory called `projects` instead?

ultrasaurus commented:
I submitted a PR for restructuring the README -- reflecting on goals in [Supply Chain Security Initiative pre-meeting notes](https://docs.google.com/document/d/1_Z8fhESffW6sPKWt8Qeu10V2A-Bg_uj5dbKJ1KXmMdU/edit#), I tried to capture the two parts

1) Raise awareness of software supply chain vulnerabilities and the implications. 
==> the catalog (as a subdirectory)
 
2) Provide best practice guidance and a catalogue of tools to mitigate vulnerabilities
==> future work

SantiagoTorres commented:
Hi @ultrasaurus , I can't seem to find the PR you mentioned. Am I missing something?

lumjjb commented on supply-chain-security/README.md:
```diff
@@ -0,0 +1,34 @@
+| [Gentoo Incident](compromises/2018/gentoo.md) | 2018    | Source Code Compromise| [1](https://wiki.gentoo.org/wiki/Project:Infrastructure/Incident_Reports/2018-06-28_Github)
+| [Unnamed Maker](compromises/2018/unnamed-maker.md) | 2018 | Publishing Infrastructure | [1](https://www.bleepingcomputer.com/news/security/microsoft-discovers-supply-chain-attack-at-unnamed-maker-of-pdf-software/) |
+| [Colourama](compromises/2018/colourama.md) | 2018 | TypoSquat | [1](https://medium.com/@bertusk/cryptocurrency-clipboard-hijacker-discovered-in-pypi-repository-b66b8a534a8), [2](https://arstechnica.com/information-technology/2018/10/two-new-supply-chain-attacks-come-to-light-in-less-than-a-week/) |
+| [Foxif/CCleaner](compromises/2017/ccleaner.md) | 2017 | Publishing Infrastructure | [1](http://blog.talosintelligence.com/2017/09/avast-distributes-malware.html) |
```
http URL has https alternative

lumjjb commented on supply-chain-security/compromises/2010/aurora.md:
Let's set a good example for caps consistency from [contributing guide](https://github.com/cncf/sig-security/blob/master/CONTRIBUTING.md) for our new additions.. I think we need to add a CI for this. Which unfortunately will probably fail for our existing repo :p

lumjjb commented on supply-chain-security/compromises/2018/gentoo.md:
```diff
@@ -0,0 +1,20 @@
+
+It seems that the attackers have been able to hack 
+the source code repository but not developer keys.
+
```
trailing whitespace

lumjjb commented on supply-chain-security/README.md:
```diff
@@ -0,0 +1,34 @@
+Software Supply Chain Compromises
+=================================
```
nit: Let's use `#` header?

ultrasaurus commented:
@SantiagoTorres I attempted to make a PR against the branch in your fork here: https://github.com/SantiagoTorres/safe/pull/1

SantiagoTorres commented:
Hi @lumjjb ! I addressed your comments. I also updated the PR wit some incidents that users submitted.

@ultrasaurus I merged your suggestion, so it's ready for another review.

Thanks!

ultrasaurus commented on supply-chain-security/compromises/README.md:
```diff
@@ -0,0 +1,42 @@
+
+| Name              | Year               | Type of compromise    | Link        |
+| ----------------- | ------------------ | ------------------    | ----------- |
+| [electron-native-notify] | 2019 | Source Code Compromise | [1](https://blog.npmjs.org/post/185397814280/plot-to-steal-cryptocurrency-foiled-by-the-npm)[2](https://komodoplatform.com/update-agama-vulnerability/)|
```
is `[electron-native-notify]` supposed to have a link?

ultrasaurus replied:
`[electron-native-notify]` supposed to have a link?

ultrasaurus commented on supply-chain-security/compromises/README.md:
```diff
@@ -0,0 +1,42 @@
+| Name              | Year               | Type of compromise    | Link        |
+| ----------------- | ------------------ | ------------------    | ----------- |
+| [electron-native-notify] | 2019 | Source Code Compromise | [1](https://blog.npmjs.org/post/185397814280/plot-to-steal-cryptocurrency-foiled-by-the-npm)[2](https://komodoplatform.com/update-agama-vulnerability/)|
+| [ShadowHammer](compromises/2019/shadowhammer.md) | 2019 | Multiple steps | [1](https://www.csoonline.com/article/3384259/asus-users-fall-victim-to-supply-chain-attack-through-backdoored-update.html), [2](https://securelist.com/operation-shadowhammer/89992/) |
```
broken links:  since README is in `compromises` directory, need to remove `compromises/` from links

ultrasaurus commented on supply-chain-security/compromises/README.md:
```diff
@@ -0,0 +1,42 @@
+| [Monju Incident](2014/monju.md) | 2014    | Publishing infrastructure| [1](https://www.contextis.com/en/blog/context-threat-intelligence-the-monju-incident)
+| [Operation Aurora](2010/aurora.md) | 2010 | Watering-hole attack | [1](https://www.wired.com/2010/03/source-code-hacks/) | 
+| [ProFTPD](2010/proftpd.md) | 2010 | Source Code Repository | [1](https://www.zdnet.com/article/open-source-proftpd-hacked-backdoor-planted-in-source-code/) |
+| [gentoo rsync compromise](2003/gentoo-rsync.md) | 2003 | Source Code Repository[1](https://archives.gentoo.org/gentoo-announce/message/7b0581416ddd91522c14513cb789f17a) |
```
This line is missing a column separator... I think it should be `Source Code Repository | [1]`

and `2003/gentoo-rsync.md`  is a broken link -- looks like the actual file has a typo in filename

SantiagoTorres commented:
@ultrasaurus thank you! I did a click-test through all the links this time (on gh, rather than locally). It should be ok now...
